routes/userRankPage.js

```diff
@@ -22,24 +22,218 @@ router.get('/', function(req, res, next) {
+            });
+        }); 
+        break;
+        case "tagJavaScript":
+        User.find({ tagJavaScript: { $gt: 0 } }).sort({ tagJavaScript:-1})
+            .exec()
+            .then(docs => {
+                res.render('userRankPage', { user: req.user, tagName : tag, users: docs, tagField: fieldTag, showTable:"True" });
+            })
+            .catch(err => {
+                console.log(err);
+                res.status(200).json({
+                    error: err
+            });
+        }); 
+        break;
+        case "tagJade":
+        User.find({ tagJade: { $gt: 0 } }).sort({ tagJade:-1})
+            .exec()
+            .then(docs => {
+                res.render('userRankPage', { user: req.user, tagName : tag, users: docs, tagField: fieldTag, showTable:"True" });
+            })
+            .catch(err => {
+                console.log(err);
+                res.status(200).json({
+                    error: err
+            });
+        }); 
+        break;
+        case "tagC":
+        User.find({ tagC: { $gt: 0 } }).sort({ tagC:-1})
+            .exec()
+            .then(docs => {
+                res.render('userRankPage', { user: req.user, tagName : tag, users: docs, tagField: fieldTag, showTable:"True" });
+            })
+            .catch(err => {
+                console.log(err);
+                res.status(200).json({
+                    error: err
+            });
+        }); 
+        break;
+        case "tagFortran":
+        User.find({ tagFortran: { $gt: 0 } }).sort({ tagFortran:-1})
+            .exec()
+            .then(docs => {
+                res.render('userRankPage', { user: req.user, tagName : tag, users: docs, tagField: fieldTag, showTable:"True" });
+            })
+            .catch(err => {
+                console.log(err);
+                res.status(200).json({
+                    error: err
+            });
+        }); 
+        break;
+        case "tagVisualBasic":
+        User.find({ tagVisualBasic: { $gt: 0 } }).sort({ tagVisualBasic:-1})
+            .exec()
+            .then(docs => {
+                res.render('userRankPage', { user: req.user, tagName : tag, users: docs, tagField: fieldTag, showTable:"True" });
+            })
+            .catch(err => {
+                console.log(err);
+                res.status(200).json({
+                    error: err
+            });
+        }); 
+        break;
+        case "tagAssembly":
+        User.find({ tagAssembly: { $gt: 0 } }).sort({ tagAssembly:-1})
+            .exec()
+            .then(docs => {
+                res.render('userRankPage', { user: req.user, tagName : tag, users: docs, tagField: fieldTag, showTable:"True" });
+            })
+            .catch(err => {
+                console.log(err);
+                res.status(200).json({
+                    error: err
+            });
+        }); 
+        break;
+       
+    }   
 });
 
 function getFieldWithTag(tag) {
```

Comment:
This function already exists in routes/question and it is exported. Therefore you can just import the function and use it instead of having duplicate methods in two different files. As an example, I imported the function in questionSpec which you can refer to